### DISCUSSION TITLE
live form update from two or more field

### DISCUSSION THREAD

**Discussion by mdnur** [filamentphp/filament#10685] [Help] [answered]
Forms\Components\DatePicker::make('check_in')
                                    ->live()
                                    ->afterOrEqual("Today")
                                    ->required(),
                                Forms\Components\DatePicker::make('check_out')
                                    ->after('today')
                                    ->live()
                                    ->required(),

                                Forms\Components\TextInput::make('numberOfDays')
                                    ->label('Number of Days')
                                    ->live()
                                    ->afterStateHydrated(function (Get $get, Set $set) {
                                        $checkIn = Carbon::parse($get('check_in'));
                                        $checkOut = Carbon::parse($get('check_out'));
                                        $numberOfDays = $checkIn->diffInDays($checkOut);
                                        $set('numberOfDays', $get('check_in'));
                                    }),

i wan to show number of days live after check_in and check_in data entry how I can show this

**Comment by leandrocfe** [comment, marked as answer]
```php
public static function numberOfDays($checkIn, $checkOut): int
{
    if (blank($checkIn) || blank($checkOut)) {
        return 0;
    }

    $checkIn = Carbon::parse($checkIn);
    $checkOut = Carbon::parse($checkOut);
    $numberOfDays = $checkIn->diffInDays($checkOut);

    return $numberOfDays;
}
```

```php
DatePicker::make('check_in')
    ->live()
    ->afterOrEqual('Today')
    ->afterStateUpdated(function ($state, Get $get, Set $set) {
        $set('numberOfDays', self::numberOfDays($state, $get('check_out')));
    })
    ->required(),

DatePicker::make('check_out')
    ->after('today')
    ->live()
    ->afterStateUpdated(function ($state, Get $get, Set $set) {
        $set('numberOfDays', self::numberOfDays($get('check_in'), $state));
    })
    ->required(),

TextInput::make('numberOfDays')
    ->label('Number of Days')
    ->default(0),
```

**Reply by mdnur** [reply]
Thank you for you ans
